scenarioo-client/app/build/scenarios-overview/scenarios-overview.component.ts

```diff
@@ -0,0 +1,239 @@
+            ])
+                .subscribe(([useCaseDiffInfo, scenarioDiffInfos]) => {
+                    this.scenarios = this.diffInfoService.getElementsWithDiffInfos(scenarios, useCaseDiffInfo.removedElements, scenarioDiffInfos, 'scenario.name');
+                }, () => {
+                    this.scenarios = this.diffInfoService.getElementsWithDiffInfos(scenarios, [], [], 'scenario.name');
+                });
+        }
+    }
+
+    resetSearchField() {
+        this.searchTerm = '';
+    }
+
+    setOrder(value: string) {
+        if (this.order === value) {
+            this.reverse = !this.reverse;
+        } else {
+            this.reverse = false;
+        }
+        this.order = value;
+    }
+
+    @HostListener('window:keyup', ['$event'])
+    keyEvent(event: KeyboardEvent) {
+        switch (event.code) {
+            case 'ArrowDown':
+                const filteredScenarios = this.filterPipe.transform(this.scenarios, this.searchTerm);
+                if (this.arrowkeyLocation < (filteredScenarios.length - 1)) {
+                    this.arrowkeyLocation++;
+                }
+                break;
+            case 'ArrowUp':
+                if (this.arrowkeyLocation > 0) {
+                    this.arrowkeyLocation--;
+                }
+                break;
+            case 'Enter':
+                this.goToScenario(this.useCaseName, this.scenario[this.arrowkeyLocation].name);
+                break;
+        }
+    }
+
+    goToScenario(useCaseName: string, scenarioName: string) {
+        this.locationService.path('/scenario/' + useCaseName + '/' + scenarioName);
+    }
+
+    goToStep(useCaseName: string, scenarioName: string) {
+        this.selectedBranchAndBuildService.callOnSelectionChange((selection) => {
+            // FIXME This could be improved, if the scenario service
+            // for finding all scenarios would also retrieve the name of the first page
+            this.scenarioResource.get(
+                {
+                    branchName: selection.branch,
+                    buildName: selection.build,
+                },
+                useCaseName,
+                scenarioName,
+            ).subscribe(
+                (scenarioResult) => {
+                    const params = this.locationService.path('/step/' + useCaseName + '/' + scenarioName + '/' + scenarioResult.pagesAndSteps[0].page.name + '/0/0');
+                },
+            );
+        });
+    }
+
+    getLabelStyle(labelName: string) {
+        if (this.labelConfigurations) {
+            this.labelConfig = this.labelConfigurations[labelName];
+            if (this.labelConfig) {
+                return {
+                    'background-color': this.labelConfig.backgroundColor,
+                    'color': this.labelConfig.foregroundColor,
+                };
+            }
+        }
+    }
+
+    collapsePanel(event: boolean) {
```

Comment:
event suggests a the type Event but it's something else. Probably same as the variable. isPanelCollapsed.